tests/system/aiplatform/test_featurestore.py

```diff
@@ -217,21 +213,107 @@ def test_batch_create_features_and_ingest_feature_values(
 
         movie_entity_type.batch_create_features(feature_configs=movie_feature_configs)
 
-        movie_entity_type.ingest_from_gcs(
+        list_movie_features = movie_entity_type.list_features()
+        assert len(list_movie_features) == 3
+
+    def test_ingest_feature_values_from_df_using_feature_time_column(
+        self, shared_state, caplog
+    ):
+
+        assert shared_state["movie_entity_type"]
+        movie_entity_type = shared_state["movie_entity_type"]
+
+        caplog.set_level(logging.INFO)
+
+        aiplatform.init(
+            project=e2e_base._PROJECT, location=e2e_base._LOCATION,
+        )
+
+        movies_df = pd.DataFrame(
+            data=[
+                {
+                    "movie_id": "movie_01",
+                    "average_rating": 4.9,
+                    "title": "The Shawshank Redemption",
+                    "genres": "Drama",
+                    "update_time": "2021-08-20 20:44:11.094375+00:00",
+                },
+                {
+                    "movie_id": "movie_02",
+                    "average_rating": 4.2,
+                    "title": "The Shining",
+                    "genres": "Horror",
+                    "update_time": "2021-08-20 20:44:11.094375+00:00",
+                },
+            ],
+            columns=["movie_id", "average_rating", "title", "genres", "update_time"],
+        )
+        movies_df = movies_df.astype({"update_time": "datetime64"})
+        feature_time_column = "update_time"
+
+        movie_entity_type.ingest_from_df(
             feature_ids=[
                 _TEST_MOVIE_TITLE_FEATURE_ID,
                 _TEST_MOVIE_GENRES_FEATURE_ID,
                 _TEST_MOVIE_AVERAGE_RATING_FEATURE_ID,
             ],
-            feature_time="update_time",
-            gcs_source_uris=_TEST_MOVIES_ENTITY_TYPE_GCS_SRC,
-            gcs_source_type="avro",
+            feature_time=feature_time_column,
+            df_source=movies_df,
             entity_id_field="movie_id",
-            worker_count=2,
+            worker_count=1,
         )
 
-        list_movie_features = movie_entity_type.list_features()
-        assert len(list_movie_features) == 3
+        assert "EntityType feature values imported." in caplog.text
```

Comment:
Is there anyway we can assert the resource has been updated based on the ingest from df?

Reply:
updated with the `read` method which produce a pd from online reading, to validate the feature values before and after ingestion, using a raw data created dataframe. This in turn validated both `read` and `ingest_from_df`method.